### ISSUE TITLE
The property '' on entity type could not be found in a SelectMany query since upgrading to EF 2.1

### ISSUE THREAD

**Issue by bartekmarnane** [dotnet/efcore#13140]
I've upgraded my app to Ef Core 2.1 and a query that used to work in EF 2.0 is now giving errors.

I use variations of the same query across multiple different tables and they are all failing with the same error now that I've upgraded.

The stacktrace is quite large, so I've only included the error and the preceding line.

```
System.InvalidOperationException: The property '' on entity type 'PromotionTopicJoin' could not be found. Ensure that the property exists and has been included in the model. at Microsoft.EntityFrameworkCore.Query.ExpressionVisitors.Internal.MemberAccessBindingExpressionVisitor.GetPropertyPath(Expression expression, QueryCompilationContext queryCompilationContext, out QuerySourceReferenceExpression querySourceReferenceExpression)
```

Here are my models:

```
public class Promotion  
{
    public int PromotionId { get; set; }   
    // I've removed the other fields  

    public virtual ICollection<PromotionTopicJoin> PromotionTopicJoins {get; set;}

}

public class PromotionTopic  
{
    public int PromotionTopicId { get; set; }     

    [MaxLength(50)]
    [Required]
    public string Name { get; set; }

    public virtual ICollection<PromotionTopicJoin> PromotionTopicJoins {get; set;}
}

public class PromotionTopicJoin
{
    public int PromotionId { get; set; }
    public Promotion Promotion { get; set; }
    public int PromotionTopicId { get; set; }
    public PromotionTopic PromotionTopic { get; set; }
}
```

Here is the query I run:

```
    return await _context.PromotionTopics
        .Where(p => p.PromotionTopicId == TopicId)
        .SelectMany(p => p.PromotionTopicJoins)
        .Select(pc => pc.Promotion)
        .Select(p => new PromotionDTO
        {
            PromotionId = p.PromotionId,
            // I've removed all the other fields
            Topics = p.PromotionTopicJoins.Select(itj => itj.PromotionTopic.Name).ToList()

        })
        .ToListAsync();
```

The error is the same with just .ToList() so it's not async related.

The problem seems to be related to SelectMany navigation.  This workaround works:

```
return await _context.Set<PromotionTopicJoin>()
    .Where(pc => pc.PromotionTopicId == TopicId)
    .Select(pc => pc.Promotion)
    .Select(p => new PromotionDTO
    {
        PromotionId = p.PromotionId,
        // I've removed all the other fields
        Topics = p.PromotionTopicJoins.Select(itj => itj.PromotionTopic.Name).ToList()    
    })
    .Where(x => x.Is_Active)
    .OrderByDescending(x => x.Created)
    .ToListAsync(); 
```

### Further technical details

FYI, my versions of Microsoft.EntityFrameworkCore and Microsoft.EntityFrameworkCore.SqlServer went from 2.0.1 (where the above query worked) to 2.1.0 (where it now fails)

**Comment by ajcvickers** [comment]
@maumar to find dupe.

**Comment by LordMike** [comment]
We're also seeing this. This is very critical :|

**Comment by ajcvickers** [comment]
@smitpatel Does this look like the correct dupe: #12575

**Comment by smitpatel** [comment]
Duplicate of #12575

**Comment by ajcvickers** [comment]
@LordMike #12575 is scheduled to be patched in the 2.1.5 release of ASP.NET Core.
